### PR TITLE
Set references.InjectAnnotations to consider also image pull secrets

### LINKED ISSUE #8027: `references.InjectAnnotations` to also consider the secrets from `pod.spec.imagePullSecrets`
**How to categorize this issue?**

/area dev-productivity usability 
/kind bug

**What happened**:
I have an extension controller that is using private OCI images, hence it uses image pull secrets for the workload in the shoot controlplane. The secret with image pull credentials is modified by https://github.com/gardener/gardener/blob/e5e7de3f5ef591293231023f99970409c2a34311/pkg/utils/kubernetes/object.go#L107

which actually makes it garbage collectable. The workload is also modified by https://github.com/gardener/gardener/blob/8215a6ca44223b89f58e025c8fe53bbff4ee235b/pkg/resourcemanager/controller/garbagecollector/references/references.go#L87, however no reference annotation is added for the secrets specified in `pod.spec.imagePullSecrets`. 

As the garbage collection controller is waiting the secret to be old enough https://github.com/gardener/gardener/blob/8215a6ca44223b89f58e025c8fe53bbff4ee235b/pkg/resourcemanager/controller/garbagecollector/reconciler.go#L81 the issue cannot be observed on a freshly created cluster, however on longing running cluster the secret do gets deleted and then the extension is declared as failed because the secret is gone and the managed resource is not healthy.  

**What you expected to happen**:
`pod.spec.imagePullSecrets` to be considered as referenced resource by default by `references.InjectAnnotations`

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- Gardener version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

/area dev-productivity usability
/kind bug

**What this PR does / why we need it**:
Set references.InjectAnnotations to consider also image pull secrets

**Which issue(s) this PR fixes**:
Fixes #8027

**Special notes for your reviewer**:

**Release note**:

```bugfix developer
`pkg/resourcemanager/controller/garbagecollector/references.InjectAnnotations` now also handles `pods.spec.imagePullSecrets`. 
```

### CODE REVIEW

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

File: pkg/resourcemanager/controller/garbagecollector/references/references.go [hunk 1]
```diff
@@ -201,6 +201,10 @@ func computeAnnotations(spec corev1.PodSpec, additional ...string) map[string]st
 		}
 	}
 
+	for _, imagePullSecret := range spec.ImagePullSecrets {
```
Comment (shafeeqes) [inline pkg/resourcemanager/controller/garbagecollector/references/references.go:204]: Can you also enhance the doc string of the `InjectAnnotations` function for this change?

Comment (vpnachev) [reply pkg/resourcemanager/controller/garbagecollector/references/references.go:204]: Done

**Review by shafeeqes — COMMENTED** [review]
Thanks!
/lgtm

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**